perfkitbenchmarker/providers/azure/azure_disk.py

```diff
@@ -53,18 +79,36 @@ class AzureDisk(disk.BaseDisk):
   _lock = threading.Lock()
   num_disks = {}
 
-  def __init__(self, disk_spec, vm_name):
+  def __init__(self, disk_spec, vm_name, machine_type):
     super(AzureDisk, self).__init__(disk_spec)
     self.host_caching = FLAGS.azure_host_caching
     self.name = None
     self.vm_name = vm_name
     self.lun = None
 
+    if self.disk_type == PREMIUM_STORAGE:
+      self.metadata = PREMIUM_STORAGE_METADATA
+    elif self.disk_type == STANDARD_DISK:
+      self.metadata = {
+          disk.MEDIA: disk.HDD,
+          disk.REPLICATION: AZURE_REPLICATION_MAP[FLAGS.azure_storage_type]
+      }
+    elif self.disk_type == disk.LOCAL:
+      if any((machine_type.startswith(prefix)
```

Comment:
Done. I called it `SdbIsSSD`, because I think it makes sense here to keep the `disk.LOCAL` check where it is and move the check about what kind of disk it is to where LOCAL_SSD_PREFIXES is defined.